libeos-payg/manager.c

```diff
@@ -1017,10 +1111,51 @@ file_load_cb (GObject      *source_object,
             }
 
           /* Set the expiry time. No other validation is needed on the loaded
-           * number, as the entire range of the type is valid. */
+           * number, as the entire range of the type is valid.
+           * The value stored on disk is in terms of wall clock time, but we
+           * want self->expiry_time_secs to be in terms of CLOCK_BOOTTIME, so
+           * we're migrating here */
           memcpy (expiry_time_secs.u8, data, sizeof (expiry_time_secs.u8));
           set_expiry_time (self, now_secs,
-                           (expiry_time_secs.u64 > now_secs) ? expiry_time_secs.u64 - now_secs : 0);
+                           (expiry_time_secs.u64 > wallclock_now_secs) ? expiry_time_secs.u64 - wallclock_now_secs : 0);
+
+          /* Delete the file since we now use clock-time and expiry-seconds instead */
+          g_file_delete_async (file, G_PRIORITY_DEFAULT, cancellable,
+                               file_load_delete_cb, g_object_ref (task));
```

Comment:
Don’t you need to call `epg_multi_task_increment()` (IIRC) if you’re adding another async call to the chain?

Reply:
Yes I think you're right, and similarly for the existing `g_file_delete_async` calls